lib/beaker/shared/fog_file_parser.rb

```diff
@@ -0,0 +1,48 @@
+require 'stringify-hash'
+
+module Beaker
+  module Shared
+    # A set of functions to read .fog files
+    module FogFileParser
+      # Read a .fog file
+      #
+      # @note Loaded .fog files may use symbols for keys.
+      #    Although not clearly documented, this *is* valid:
+      #    https://www.rubydoc.info/gems/fog-core/1.42.0/Fog#credential-class_method
+      #    https://github.com/fog/fog-core/blob/7865ef77ea990fd0d085e49c28e15957b7ce0d2b/spec/utils_spec.rb#L11
+      #
+      # @param fog_file_path [String] dot fog path. Overridden by
+      # @return [StringifyHash] credentials stored in fog_file_path
+      # @raise [ArgumentError] when the specified credential file is invalid, describing the reson
+      def parse_fog_file(fog_file_path = '~/.fog', credential = :default)
```

Comment:
I think changing credential => credentials_key might make this more readable (b/c you're getting the credentials from that key). The function name should also be changed to `parse_credentials_from_fog_file` (or something similar), b/c you're parsing the credentials here, not the entire fog file.

Reply:
I have [re-used the naming convention from fog-core](https://github.com/fog/fog-core/blob/f6ef6c3d9458bc235be57100acdced264e85e52a/lib/fog/core/credentials.rb#L35), but am not attached to it so the name change sounds good. Actually, looking at this again, this isn't so much a file parser as it is a generic fog credential provider (note the preference for environment variables, as described in the fog spec), so I'll rename it.